packages/sdk/python/src/relayfile/on_write.py

```
@@ -0,0 +1,378 @@
+            if event.operation not in registration.operations or not path_matches(registration.pattern, event.path):
+                continue
+            pattern_lock = self._pattern_locks.setdefault(registration.pattern, threading.Lock())
+            with pattern_lock:
+                self._run_handler(registration, event)
+
+    def _run_handler(self, registration: _Registration, event: WriteEvent) -> None:
+        try:
+            result = registration.handler(event)
+            if hasattr(result, "__await__"):
+                asyncio.run(result)
+        except Exception as exc:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)